crates/common/tedge_config/src/tedge_config_cli/tedge_config_dto.rs

```diff
@@ -112,6 +114,17 @@ pub(crate) struct AwsConfigDto {
 #[serde(deny_unknown_fields)]
 pub(crate) struct MqttConfigDto {
     pub(crate) port: Option<u16>,
+
+    pub(crate) client_address: Option<String>,
+
+    // When connecting to a host, port 0 is invalid. When binding, however, port 0 is accepted and
+    // understood by the system to dynamically assign any free to the process. The process then
+    // needs to take notice of what port it received, which I'm not sure if we're doing. If we don't
+    // want to allow binding to port 0, then we can also use `NonZeroU16` there as well, which
+    // because it can never be 0, can make the `Option` completely free, because Option can use 0x0000
+    // value for the `None` variant.
```

Comment:
I don't see any use for this `0` port in the context of thin-edge. We need well-known address and port to be used by the child devices.

I would not bother with an optimization using `0` for `None`.

Reply:
What I meant is that currently we allow the user to set `MqttPortSetting` or `HttpPortSetting` to 0, which can lead to things being broken. It is a user error, so it doesn't matter that much, but by using this type here, everybody reading this code will know that port can't be 0, and it forces us to handle the case when the user tries to use 0, which isn't valid. The optimization is just a free side effect. As you said, we need a well-known port, so `0` is invalid, so I'll keep the type as it is. If we want to forbid using 0 for external port as well, which we currently don't then perhaps it would be good to change it as part of this PR as well?

As for the comment though, I've written it to explain why `client_port` and `external_port` use different types for the purposes of the review. Let me know if I should remove it.